bin/main.ml

```diff
@@ -14,27 +10,72 @@ let get_contents = function
 
 let parse_file file = get_contents file |> Parse.from_string
 
-let main files =
-  match files with
-  | [] ->
-    let ast = parse_file "-" in
-    ignore @@ Interpret.start ast
-  | _ ->
-    ignore
-    @@ List.fold_left
-         (fun state file ->
-           let ast = Parse.from_file ~filename:file in
-           Some (Interpret.start ?state ast) )
-         None files
+type prover =
+  | Z3_prover
+  | Colibri2_prover
 
-let files =
-  let doc = "source files" in
-  Arg.(value & pos_all non_dir_file [] & info [] ~doc)
+let prover_conv =
+  Cmdliner.Arg.conv
+    ( (fun str ->
+        try
+          Ok
+            ( match str with
+            | "Z3" | "z3" -> Z3_prover
+            | "Colibri2" | "colibri2" -> Colibri2_prover
+            | _ ->
+              failwith (Fmt.str "Unsupported prover %s, try Z3 or Colibri2" str)
+            )
+        with Failure s -> Error (`Msg s) )
```

Comment:
Why do we throw the exception here? Wouldn't putting the `Error ('Msg (Fmt.str ...` in the failwith case be more straightforward? I don't usually like doing control-flow with exceptions unless there is a gain to be had, but let me know what you :)